src/Controller/Frontend/SearchController.php

```diff
@@ -37,12 +37,23 @@ public function __construct(Config $config, Environment $twig, TemplateChooser $
     public function search(ContentRepository $contentRepository, Request $request): Response
     {
         $page = (int) $request->query->get('page', 1);
+        $search = $request->get('search', '');
+        $amountPerPage = $this->config->get('general/listing_records');
 
         // @todo implement actual Search Engine
-        $records = $contentRepository->findForListing($page);
+        if (! empty($search)) {
+            $records = $contentRepository->searchNaive($search, $page, $amountPerPage);
+        } else {
+            $records = null;
```

Comment:
it should be empty array, otherwise pagerfanta etc. will break ;)

Reply:
also add a test what will happen if query is empty

Reply:
I _think_ `null` is correct, and it should be `?Pagerfanta $records`. I've added a test to illustrate. :-)

![Schermafbeelding 2019-03-27 om 16 08 14](https://user-images.githubusercontent.com/1833361/55087573-98d0e480-50aa-11e9-9890-a35ac7be028b.png)

Reply:
yes, it gives an exception because this line returns null instead of empty array

Reply:
No, it _breaks_ as soon as that line returns `[]`. If the method accepts only `Pagerfanta $records`, it does break if you're trying to pass in `[]`.. 

On the other hand, `?Pagerfanta $records` allows you to pass in `null` and handle it properly.

Reply:
Right, so let's return empty Pagerfanta then, to keep context interface clean.

Reply:
Ok, sounds good. Done.